bank_statement_parse/parserlib.py

```diff
@@ -0,0 +1,228 @@
+        self.error_message = False  # error message for interaction with user
+        self.storno_retry = False
+        # If True, make cancelled debit eligible for a next direct debit run
+        self.data = ''  # Raw data from which the transaction has been parsed
+
+
+class BankStatement(dict):
+    """A bank statement groups data about several bank transactions."""
+
+    @property
+    def statement_id(self):
+        """property getter"""
+        return self['name']
+
+    def _set_transaction_ids(self):
+        """Set transaction ids to statement_id with sequence-number."""
+        subno = 0
+        for transaction in self['transactions']:
+            subno += 1
+            transaction['unique_import_id'] = (
+                self.statement_id + str(subno).zfill(4))
+
+    @statement_id.setter
+    def statement_id(self, statement_id):
+        """property setter"""
+        self['name'] = statement_id
+        self._set_transaction_ids()
+
+    @property
+    def local_account(self):
+        """property getter"""
+        return self['account_number']
+
+    @local_account.setter
+    def local_account(self, local_account):
+        """property setter"""
+        self['account_number'] = local_account
+
+    @property
+    def local_currency(self):
+        """property getter"""
+        return self['currency_code']
+
+    @local_currency.setter
+    def local_currency(self, local_currency):
+        """property setter"""
+        self['currency_code'] = local_currency
+
+    @property
+    def start_balance(self):
+        """property getter"""
+        return self['balance_start']
+
+    @start_balance.setter
+    def start_balance(self, start_balance):
+        """property setter"""
+        self['balance_start'] = start_balance
+
+    @property
+    def end_balance(self):
+        """property getter"""
+        return self['balance_end']
+
+    @end_balance.setter
+    def end_balance(self, end_balance):
+        """property setter"""
+        self['balance_end'] = end_balance
+        self['balance_end_real'] = end_balance
+
+    @property
+    def date(self):
+        """property getter"""
+        return self['date']
+
+    @date.setter
+    def date(self, date):
+        """property setter"""
+        self['date'] = date
+
+    def create_transaction(self):
+        """Create and append transaction.
+
+        This should only be called after statement_id has been set, because
+        statement_id will become part of the unique transaction_id.
+        """
+        transaction = BankTransaction()
+        self['transactions'].append(transaction)
+        # Fill default id, but might be overruled
+        transaction['unique_import_id'] = (
+            self.statement_id + str(len(self['transactions'])).zfill(4))
+        return transaction
+
+    def __init__(self):
+        super(BankStatement, self).__init__()
+        self['transactions'] = []
+        self.statement_id = ''
+        self.local_account = ''
```

Comment:
Initializing local_account here is not necessary since the property getter/setter map it to account_number.